modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/persistence/entity/data/impl/MybatisCaseInstanceDataManagerImpl.java

```diff
@@ -224,4 +225,26 @@ protected void setSafeInValueLists(CaseInstanceQueryImpl caseInstanceQuery) {
             }
         }
     }
+
+    @Override
+    public long countChangeTenantIdCmmnCaseInstances(String sourceTenantId, boolean onlyInstancesFromDefaultTenantDefinitions) {
+        String defaultTenantId = getCmmnEngineConfiguration().getDefaultTenantProvider().getDefaultTenant(sourceTenantId, ScopeTypes.CMMN, null);
```

Comment:
I am not sure that using the default tenant provider in such a low level is OK. The DataManagers are usually should not do any business specific logic. The default tenant provider in my opinion is something more high level.

This comment is valid for all the places the default tenant provider is used in the data managers.

Reply:
I have moved this logic to the abstract class that each data manager expands. I have seen that something similar is done for the getIdGenerator() method.